test/core/security/secure_endpoint_test.c

```diff
@@ -44,9 +44,11 @@ static grpc_endpoint_test_fixture secure_endpoint_create_fixture_tcp_socketpair(
   tsi_frame_protector *fake_write_protector =
       tsi_create_fake_frame_protector(NULL);
```

Comment:
If `use_zero_copy_frame_protector` is false, don't we need to avoid creating the old-style frame protectors?

Reply:
Well, there is encrypted_leftover is still created by old-style frame protector (for convenience reason, otherwise, I have to create encrypt_leftover using two different methods which has same result). Encrypted leftover frame should be interoperable regardless of whether we use old frame protector or zero-copy protector to create. We can clean up the secure_endpoint_test when we completely get rid of old-style frame protector. What do you think?

Reply:
My concern was that we might have the same memory leak in reverse -- i.e., if `use_zero_copy_frame_protector` is true, do we still destroy the old-style frame protector?  If so, this is fine.

Reply:
In the revised code, we will pass both old-style protector and zero-copy protector (which could be NULL) into secure_endpoint creation. In secure_endpoint.c, destroy will destruct both old-style protector and zero-copy protector. It is just that if zero-copy protector is not NULL, then old-style protector is never used, otherwise, old-style protector is used.

Let's wait for all tests to finish.

Once we fully migrate to zero-copy protector, the code would be cleaner.